src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp

```diff
@@ -774,18 +788,25 @@ std::shared_ptr<ov::npuw::CompiledModel> ov::npuw::CompiledModel::deserialize(
 
     // Initialize weights stream if weightless flow
     std::string weights_path;
+    std::shared_ptr<const ov::Model> model_ptr;
     if (is_weightless) {
-        NPUW_ASSERT(properties.find(ov::weights_path.name()) != properties.end() &&
-                    "There is no WEIGHTS_PATH set in properties but the blob was exported as weightless!");
-        weights_path = properties.at(ov::weights_path.name()).as<std::string>();
+        if (properties.find(ov::weights_path.name()) != properties.end()) {
+            weights_path = properties.at(ov::weights_path.name()).as<std::string>();
+        } else if (properties.find(ov::hint::model.name()) != properties.end()) {
```

Comment:
As it's set by the Core, we need to add it to the supported properties in the NPU plugin as we did with weights_path

Reply:
Done